doc/src/tutorials/intro-tutorial/basic_operations.rst

```diff
@@ -129,13 +129,22 @@ To evaluate a numerical expression into a floating point number, use
     >>> expr.evalf()
     2.82842712474619
 
+.. raw:: latex
+
+   \begingroup
+   \sphinxsetup{verbatimforcewraps}
```

Comment:
Should we just add this to the latex configuration? I'm sure this isn't the only example that overflows in the PDF.

Reply:
It could be added to global configuration like this:
```
latex_elements = {
    ...other things...,
    'sphinxsetup': 'verbatimforcewraps',
}
```

Normally long code lines are wrapped without this setting satisfactorily because break points are found at least at spaces and at various non-letter punctuation.  The setting is needed for things such as this case of a long string of digits, one can also imagine DNA sequence of such types of things.  It causes Sphinx to use more complex LaTeX macro layer which remaps all mark-up applied by Pygments styling to apply character per character, inserting then breakpoints everywhere.

If you can get a hold of the LaTeX buid log file, checking for "overfull \hbox" should help locate other cases and see how frequent or numerous they are  I tried perusing the PDF but did not immediately find other instances, now, I am only a human not an AI... (which would immediately locate in many thousands pages the strings overflowing into margins...).

Reply:
There is a build log available if you build the pdf (see https://docs.sympy.org/latest/contributing/new-contributors-guide/build-docs.html#pdf-documentation). Or you could modify the GitHub actions to put the log as an artifact. I seem to remember that there are quite a few overfull \hbox errors in the log the last time I looked at it.

Reply:
Ok, I managed to build locally after installing a Chrome Legacy for my old macos.  The svg to pdf conversions failed despite chromium being in expected place but a least this did not cause a fatal error as with my Inkscape 0.92.2 and I could then try out the xelatex builds.  Of course it would be probably annoying to add pdf's to the git, but if these images are not dynamic, then perhaps you may consider using glob pattern in rst sources (not sure how/if it works with myST) so that Sphinx knows to use the pre-prepared pdf's for LaTeX. (but sometimes pdf inclusion has [problems](https://tex.stackexchange.com/questions/106964/could-not-insert-pdf-graphics) with xelatex).

The Makefile should add `logo` as dependency to `latex` targets, because as I did not build the html docs, the `_build/logo` was missing, and the produced LaTeX file expects to find some `png`'s in-there and build fails without them.

Turns out I did the build using my branch for PR #26866, so the end of log was already missing probably dozens (or dozens of dozens) `Overfull \hbox` generated from general Index, but I did find some interesting ones, but only 3 of the type of the pi digits overflow. (I discuss this in more detail next).  Before this, I had checked Sphinx and now I remember that the extra overhead added by `verbatimforcewraps` is not excessive: it adds an extra "measuring phase" to all renderings of codelines, which may not be that time costly anyhow (I did not time this because the build takes time and I had to address other issues first such as the logo business), and if that phase detects an overflow (which for Sympy is very rare) then it does the more hacky things I alluded too (the whole business being already quite deep core TeX...).

In my build (which as i said is missing some pdf's, so page numbering maybe off) the few instances similar to pi digits were on pages 1380, 1838, and 2285 (and more of same type a bit after I think).  Here are screenshots

page 1380
![Capture d’écran 2024-07-30 à 11 11 32](https://github.com/user-attachments/assets/e086ada6-4b72-4f9d-ac5f-66993cc421e1)

page 1838
![Capture d’écran 2024-07-30 à 11 12 44](https://github.com/user-attachments/assets/b64f551f-0d59-4500-847b-40dadd50c876)

page 2235
![Capture d’écran 2024-07-30 à 11 25 52](https://github.com/user-attachments/assets/a8291d7e-8313-4956-aa91-fe6f2e12f315)

The last case is irreparable.  The nature of the thing (Maple like output) can only be wrapped by human intervention, impossible to do this automatically at Sphinx level.  The first two would be solved by local or global `verbatimforcewraps`.

When checking `Overfull \hbox` the line number in LaTeX file is indicated which allows to locate source too.  The page number in PDF is most of the time the next bracketed one, but sometimes it is actually on next page, for example when a tabular is too long to fit on page and is moved to next one during latex "output routine".  The log line indicated the amount of overfull and you only need to worry (for letter sized paper) when it exceeds 70pt.  Above that you do lose characters beyond page limits.

Some overfull are located in a few tables, and probably could be fixed by suitable usage of Sphinx  [``tabularcolumns`` directive](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-tabularcolumns).  Only two or three tables are involved the worst one is this but I have lost on which page, the other tables had more benign problems.
![Capture d’écran 2024-07-30 à 11 17 42](https://github.com/user-attachments/assets/6cb1d52d-c2a1-42a9-ad91-a6449ccdb511)

Then they are overfull caused by too long displayed equations. This can only be fixed in sources (I don't know of a robust LaTeX solution to automatically break lines in math displays, a package breqn I think exist but causes problems in some circumstances, perhaps worth a try, not tested).

Here is an example (5.8. Topics page 837)
![Capture d’écran 2024-07-30 à 11 03 37](https://github.com/user-attachments/assets/87e23847-f4f1-4915-bd9d-ef50550a08f1)

There are a few other math display problems not very numerous.

Then there is sometimes a problem with an hyperlink which overflows in margin, normally hyperref knows how to break it but it may not find suitable breakpoints (and I think never break at slashes), but this is important only for print version, as numeric PDF will have the correct target on hover with mouse.

The last class of problems may need some ticket opening at https//github.com/sphinx-doc/sphinx.
I located a dozen cases like this one (one of the worst):

![Capture d’écran 2024-07-30 à 11 22 05](https://github.com/user-attachments/assets/ad0ed6a7-5549-491c-a71d-3bd005c5151e)

A few more:
![Capture d’écran 2024-07-30 à 11 10 45](https://github.com/user-attachments/assets/f07af736-3207-4702-9c4c-b8635592a714)

![Capture d’écran 2024-07-30 à 11 07 30](https://github.com/user-attachments/assets/4bcb270f-356b-4ef7-a824-941296063ae7)

![Capture d’écran 2024-07-30 à 11 20 29](https://github.com/user-attachments/assets/1570aa65-852f-40bc-9424-6cd639b94d4b)

Such problemes deserve if you have time opening a ticket at Sphinx reusing one of abvoe images.

Finally, some Missing characters are reported, see #26877.

Reply:
I intend to become much more familiar with Sympy in future and pretty printed output looks like a pretty cool feature...

Reply:
Tomorrow I will have a look at the few tables where I identified too narrow columns.  Suitable usage of `tabularcolumns` Sphinx directive will solve that.

Reply:
> Should we just add this to the latex configuration? I'm sure this isn't the only example that overflows in the PDF.

Although I found only two further examples with overflowing digits I agree it is a bit painful to have to maintain raw latex directive in sources.  So I am going to revert what I did for the initial spotted case, the one of pi digits, and add the `verbatimforcewraps` in conf.py.  Tomorrow I will test if I see any significant increase in build time.

Reply:
(I don't know if Sympy process is via squash/rebase during merge to master, but I am expecting it so I do not worry on my accumulating small commits).  I plan to push to this branch some more commit improving tabulars in PDF.  And I will raise at some point an upstream Sphinx issue about documenting objects with no room left on right side for parameters in the signature. It seems we should in those cases shift to under the long object name, simply, regaining ample room for arguments (especially if they come with type annotations this is quite needed).

Reply:
> don't know if Sympy process is via squash/rebase during merge to master, but I am expecting it so I do not worry on my accumulating small commits

See https://github.com/sympy/sympy/issues/18548